scripts/regression_test/extension_regression_test.yml

```diff
@@ -0,0 +1,191 @@
+  strategy:
+    maxParallel: 12
+    matrix:
+      instance1:
+        Instance_idx: 1
+      instance2:
+        Instance_idx: 2
+      instance3:
+        Instance_idx: 3
+      instance4:
+        Instance_idx: 4
+      instance5:
+        Instance_idx: 5
+      instance6:
+        Instance_idx: 6
+      instance7:
+        Instance_idx: 7
+      instance8:
+        Instance_idx: 8
+      instance9:
+        Instance_idx: 9
+      instance10:
+        Instance_idx: 10
+      instance11:
+        Instance_idx: 11
+      instance12:
+        Instance_idx: 12
+  pool:
+    name: ${{ variables.ubuntu_pool }}
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '3.10'
+      displayName: "Use Python 3.10"
+  - task: AzureCLI@1
+    displayName: 'checkout cli and extension repo'
+    inputs:
+      azureSubscription: 'Azure CLI release'
+      scriptLocation: inlineScript
+      inlineScript: |
+        set -ev
+        pwd
+        git clone https://github.com/Azure/azure-cli-extensions.git
+        cd azure-cli-extensions
+
+        # git config
+        GITHUB_TOKEN=$(az keyvault secret show --vault-name kv-azuresdk --name azclibot-pat --query value -otsv)
+        git config --global user.email "AzPyCLI@microsoft.com"
+        git config --global user.name "Azure CLI Team"
+        git remote add azclibot https://azclibot:${GITHUB_TOKEN}@github.com/azclibot/azure-cli-extensions.git
+
+        git fetch azclibot
+        git checkout -b regression_test_$(Build.BuildId) azclibot/regression_test_$(Build.BuildId)
+  - template: ../../.azure-pipelines/templates/azdev_setup.yml
+    parameters:
+      CLIExtensionRepoPath: ./azure-cli-extensions
+  - bash: |
+      set -ev
+
+      source env/bin/activate
+      cd azure-cli-extensions
+
+      az login -u $(CLI_LIVE_TEST_ACCOUNT) -p "$(CLI_LIVE_TEST_PASSWORD)"
+      az account set -s 0b1f6471-1bf0-4dda-aec3-cb9272f09590
+
+      python ../scripts/ci/automation_full_test.py "12" "$(Instance_idx)" "latest" "" "True" "extension"
+    displayName: "Rerun tests"
+  - bash: |
+      publishErrorModules='false'
+      if [[ -f '/mnt/vss/.azdev/env_config/mnt/vss/_work/1/s/env/test_results_error_modules_$(Instance_idx).txt' ]]; then
+        publishErrorModules='true'
+      fi
+      echo "##vso[task.setvariable variable=publishErrorModules]$publishErrorModules"
+
+      publishFailureTests='false'
+      if [[ -f '/mnt/vss/.azdev/env_config/mnt/vss/_work/1/s/env/test_results_failure_tests_$(Instance_idx).txt' ]]; then
+        publishFailureTests='true'
+      fi
+      echo "##vso[task.setvariable variable=publishFailureTests]$publishFailureTests"
+    condition: succeededOrFailed()
+    displayName: "Prepare publishing results"
+  - task: PublishTestResults@2
+    condition: succeededOrFailed()
+    inputs:
+      testResultsFiles: '/mnt/vss/.azdev/env_config/mnt/vss/_work/1/s/env/test_results_*.xml'
+      testRunTitle: 'CLI Regression test results of instance $(Instance_idx)'
```

Comment:
It will be great if we can adopt this `PublishTestResults` task in 
[Azure.azure-cli Full Test](https://dev.azure.com/azclitools/public/_build?definitionId=62&_a=summary) pipeline.